### PR TITLE
Add support to build FFU for NXP iMX8M EVK

### PR DESCRIPTION
**Pull request by neilsh-msft**
Add package and FFU imaging files for NXP MCIMX8MEVK board (ARM64)
Add U-Boot and UEFI binary images for MCIMX8MEVK board
Rationalize output directory structure to PLATFORM\Configuration:  (ARM|ARM64) \ (Debug|Release)
Check for PLATFORM setting in BuildBSP.bat and remove unused IMX[6|7]-[Debug|Release] Configuration settings.

### CODE REVIEW

File: build/board/NXPEVK_iMX8M_4GB/NXPEVK_iMX8M_4GB_ProductionOEMInput.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  <Feature>IOT_CRT140</Feature>
+  <Feature>IOT_POWERSHELL</Feature>
+  <Feature>IOT_POWER_SETTINGS</Feature>
+  <Feature>IOT_ENABLE_ADMIN</Feature>
+  <Feature>IOT_TOOLKIT</Feature>
+  <Feature>IOT_WDTF</Feature>
+  <Feature>IOT_ENABLE_TESTSIGNING</Feature>
+  <Feature>IOT_UMDFDBG_SETTINGS</Feature>
+  <Feature>IOT_USBFN_CLASS_EXTENSION</Feature>
+<!--  <Feature>IOT_ONBOARDING_APP</Feature> -->
```
Comment (christopherco) [inline build/board/NXPEVK_iMX8M_4GB/NXPEVK_iMX8M_4GB_ProductionOEMInput.xml:51]: Do we want to keep all these features commented out or just remove them wholesale?

File: build/tools/BuildImage.bat [hunk 2]
```diff
@@ -87,16 +87,16 @@ if "%SolutionDir%"=="" (
...
 REM
-set BSPPKG_DIR=%REPO_BUILD_ROOT%\solution\iMXPlatform\Build\FFU\bspcabs
+set BSPPKG_DIR=%REPO_BUILD_ROOT%\solution\iMXPlatform\Build\FFU\bspcabs\%PLATFORM%\%Configuration%
 set PKG_VER=1.0.0.0
 if not exist %BSPPKG_DIR% ( mkdir %BSPPKG_DIR% )
 
-if not exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM (dir /S /B %REPO_BUILD_ROOT%\solution\iMXPlatform\Build\%Configuration%\ARM\*.cab > filelist.txt)
-if exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM (dir /S /B %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM\*.cab > filelist.txt)
+if not exist %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration% (dir /S /B %REPO_BUILD_ROOT%\solution\iMXPlatform\Build\%PLATFORM%\%Configuration%\*.cab > filelist.txt)
+if exist %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration% (dir /S /B %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration%\*.cab > filelist.txt)
```
Comment (hbeberman) [inline build/tools/BuildImage.bat]:
```suggestion
if exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\%PLATFORM% (dir /S /B %REPO_BUILD_ROOT%\..\..\b\%Configuration%\%PLATFORM%\*.cab > filelist.txt)
```

File: build/tools/BuildImage.bat [hunk 3]
```diff
@@ -87,16 +87,16 @@ if "%SolutionDir%"=="" (
...
 echo Export packages to one directory
 REM
-set BSPPKG_DIR=%REPO_BUILD_ROOT%\solution\iMXPlatform\Build\FFU\bspcabs
+set BSPPKG_DIR=%REPO_BUILD_ROOT%\solution\iMXPlatform\Build\FFU\bspcabs\%PLATFORM%\%Configuration%
 set PKG_VER=1.0.0.0
 if not exist %BSPPKG_DIR% ( mkdir %BSPPKG_DIR% )
 
-if not exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM (dir /S /B %REPO_BUILD_ROOT%\solution\iMXPlatform\Build\%Configuration%\ARM\*.cab > filelist.txt)
-if exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM (dir /S /B %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM\*.cab > filelist.txt)
+if not exist %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration% (dir /S /B %REPO_BUILD_ROOT%\solution\iMXPlatform\Build\%PLATFORM%\%Configuration%\*.cab > filelist.txt)
```
Comment (hbeberman) [inline build/tools/BuildImage.bat]:
```suggestion
if not exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\%PLATFORM% (dir /S /B %REPO_BUILD_ROOT%\solution\iMXPlatform\Build\%PLATFORM%\%Configuration%\*.cab > filelist.txt)
```

File: build/tools/BuildImage.bat [hunk 4]
```diff
@@ -87,16 +87,16 @@ if "%SolutionDir%"=="" (
...
+if exist %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration% (dir /S /B %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration%\*.cab > filelist.txt)
 
 REM Append the %BOARD_NAME%Package folder to the end of the scrape list so the correct SV.PlatExtensions.UpdateOS.cab is in bspcabs
-if not exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM (dir /S /B %REPO_BUILD_ROOT%\solution\iMXPlatform\Build\%Configuration%\ARM\%BOARD_NAME%Package\*.cab >> filelist.txt)
-if exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM (dir /S /B %REPO_BUILD_ROOT%\..\..\b\%Configuration%\ARM\%BOARD_NAME%Package\*.cab >> filelist.txt)
+if not exist %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration% (dir /S /B %REPO_BUILD_ROOT%\solution\iMXPlatform\Build\%PLATFORM%\%Configuration%\%BOARD_NAME%Package\*.cab >> filelist.txt)
```
Comment (hbeberman) [inline build/tools/BuildImage.bat]:
```suggestion
if not exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\%PLATFORM% (dir /S /B %REPO_BUILD_ROOT%\solution\iMXPlatform\Build\%PLATFORM%\%Configuration%\%BOARD_NAME%Package\*.cab >> filelist.txt)
```

File: build/tools/BuildImage.bat [hunk 5]
```diff
@@ -87,16 +87,16 @@ if "%SolutionDir%"=="" (
...
+if exist %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration% (dir /S /B %REPO_BUILD_ROOT%\..\..\b\%PLATFORM%\%Configuration%\%BOARD_NAME%Package\*.cab >> filelist.txt)
```
Comment (hbeberman) [inline build/tools/BuildImage.bat]:
```suggestion
if exist %REPO_BUILD_ROOT%\..\..\b\%Configuration%\%PLATFORM% (dir /S /B %REPO_BUILD_ROOT%\..\..\b\%Configuration%\%PLATFORM%\%BOARD_NAME%Package\*.cab >> filelist.txt)
```

**Review by hbeberman — CHANGES_REQUESTED** [review]
Some of the changes in BuildImage.bat have broken the build server.

The server is always generating its binary output folder as \b\%Configuration%\%Platform%, so when the script checks to see if it needs to scrape the cabs from there or from within the source folders itself it's assuming it's a local build and failing to copy the cabs for FFU generation.

File: build/tools/BuildImage.bat (hunk 3 above)
Comment (christopherco) [reply build/tools/BuildImage.bat]: Recommend adding a comment explaining which config is specific to our CI build server for future reference.

File: build/board/NXPEVK_iMX8M_4GB/NXPEVK_iMX8M_4GB_ProductionOEMInput.xml (hunk 1 above)
Comment (neilsh-msft) [reply build/board/NXPEVK_iMX8M_4GB/NXPEVK_iMX8M_4GB_ProductionOEMInput.xml:51]: The IOT_ONBOARDING_APP isn't available on the ARM64 platform yet.  The others are.  We don't include them for the other boards in Kayla, but they are included in the .xml files in the OS tree.

Comment (christopherco) [reply build/board/NXPEVK_iMX8M_4GB/NXPEVK_iMX8M_4GB_ProductionOEMInput.xml:51]: Okay, sounds fine to leave IOT_ONBOARDING_APP commented out then.

**Pull request merged**